### PR TITLE
WorkItem#Solve Example

### PR DESCRIPTION
**Pull request by JHamide**

### CODE REVIEW

**Review by mohammadKarimi — CHANGES_REQUESTED** [review]
why you change the whole code base. 
it is not good idea to pass action<sting> in this case.

**Comment by JHamide** [comment]
I've made the requested changes, please review again.

File: Examples/Ticketing.cs [hunk 1]
```diff
@@ -1,14 +1,56 @@
...
+public interface ITicketService
+{
+    public void GetTicket(string type);
+}
+
+public interface IMovieTicketService : ITicketService
+{
+}
+
+public interface IConcertTicketService : ITicketService
```
Comment (mohammadKarimi) [inline Examples/Ticketing.cs:26]: what is the purpose of this empty interface.

File: Examples/Ticketing.cs [hunk 2]
```diff
@@ -1,14 +1,56 @@
...
+            new FlightTicketService().GetTicket(ticketType);
         else
             throw new ArgumentException("Invalid ticket type.");
     }
 }
+
+
+public interface ITicketService
+{
+    public void GetTicket(string type);
```
Comment (mohammadKarimi) [inline Examples/Ticketing.cs:19]: your method is GetTicket but return type is void. it is not good practice of readibilty.

File: Examples/Ticketing.cs [hunk 3]
```diff
@@ -1,14 +1,56 @@
 public class TicketService
 {
-    public ... Get(string ticketType)
+    public void Get(string ticketType)
```
Comment (mohammadKarimi) [inline Examples/Ticketing.cs:3]: same as previous comment.

**Review by mohammadKarimi — CHANGES_REQUESTED** [review]
please follow my comments.

File: Examples/Ticketing.cs (hunk 2 above)
Comment (JHamide) [reply Examples/Ticketing.cs:19]: I believed that if additional methods were needed for this class in the future, I could define them within the interface and then implement them in the class.

File: Examples/Ticketing.cs (hunk 1 above)
Comment (JHamide) [reply Examples/Ticketing.cs:26]: I believed that if additional methods were needed for this class in the future, I could define them within the interface and then implement them in the class.

**Comment by JHamide** [comment]
I corrected it. Thanks a lot.

File: Examples/NotificationService.cs [hunk 4]
```diff
@@ -1,15 +1,37 @@
 public class NotificationService()
 {
-    public ... Get(string type)
+    public void Get(string type)
```
Comment (mohammadKarimi) [inline Examples/NotificationService.cs:3]: same as previous comment.

Comment (mohammadKarimi) [reply Examples/NotificationService.cs:3]: you have to change your return type, since we have a method to Get a new instance.

**Review by mohammadKarimi — CHANGES_REQUESTED** [review]

**Pull request closed without merging**